rr_viz/src/ross/mission_editor/mission_editor.py

```diff
@@ -0,0 +1,201 @@
+        self.h_layout_command.addWidget(self.go_to_all_button)
+        self.h_layout_command.addWidget(self.send_mission_button)
+
+        self.v_layout.addLayout(self.h_layout_command)
+
+        self.setLayout(self.v_layout)
+
+        self.bt_action = BuildBTAction()
+        self.mb_action = WaypointMoveBaseAction()
+        
+        # Setup state checker:
+        self.bt_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.bt_action.is_connected,  self.set_enable_send_mission, Hz=1./3.)
+        self.bt_state_checker.start()
+        self.mb_state_checker = rr_qt_helper.StateCheckerTimer(
+            self.mb_action.is_connected,  self.set_enable_go_to_buttons, Hz=1./3.)
+        self.mb_state_checker.start()
+
+    def robot_pose_cb(self, msg):
+        self.robot_pose = msg
+
+    def spawn_waypoint(self, pwcs_msg):
+        msg = TaskWaypointMsg()
+        pose_stamped = PoseStamped()
+        pose_stamped.header = pwcs_msg.header
+        pose_stamped.pose = pwcs_msg.pose.pose
+        msg.pose_stamped = pose_stamped
+        self.waypoint_list.append(self.waypoint_list.new_waypoint(msg))
+
+    def add_here(self):
+        if self.robot_pose:
+            msg = PoseWithCovarianceStamped()
+            msg.header = self.robot_pose.header
+            msg.pose.pose = self.robot_pose.pose
+            self.spawn_waypoint_signal.emit(msg)
+        else:
+            rospy.logwarn("Robot pose not connected")
+
+    def duplicate(self):
+        if self.waypoint_list.get_selected_wp():
+            self.waypoint_list.duplicate(self.waypoint_list.get_selected_wp())
+
+    def delete(self):
+        if self.waypoint_list.get_selected_wp():
+            self.waypoint_list.remove(self.waypoint_list.get_selected_wp())        
+
+    def delete_all(self):
+        for _ in range(self.waypoint_list.len()):
+            self.waypoint_list.remove(
+                self.waypoint_list.get_wp(self.waypoint_list.len()-1))
+
+    def save(self):
+        mission_name, ok = QInputDialog.getText(self, "Mission file name", "Specify file name to save mission:")
+
+        if ok:
+            if mission_name[-5:] == ".yaml":
+                mission_name = mission_name[0:-5]
+
+            mission_name = string.replace(mission_name, '.', '_')
+            mission_name = mission_name + ".yaml"
+                        
+            save_path = file_management.get_mission_files_dir() + "/" + mission_name            
+            self.waypoint_list.saveToPath(save_path, "Mission" + str(rospy.Time.now()))
+
+    def load(self):
+        mission_files = file_management.get_files(file_management.get_mission_files_dir(), ".yaml")
+        
+        mission_file, ok = QInputDialog.getItem(self, "Select mission to load", "Available missions:", mission_files, 0, False)
+        
+        if ok:
+            load_path = file_management.get_mission_files_dir() + "/" + mission_file
+            if load_path[-5:] != ".yaml":
```

Comment:
here as well I would use .endswith, to handle weird cases if the filename is short